include/nbl/system/IAsyncQueueDispatcher.h

```diff
@@ -19,10 +19,20 @@ namespace impl
             // wait on this for result to be ready
             std::condition_variable cvar;
```

Comment:
do you ever use `wait_for` or `wait_until` on this `cvar` ?

Reply:
nope

Reply:
then under c++20 you can replace it with just waiting on the atomic itself ;)

Reply:
just make sure you note it down that with C++20 we could get rid of both the `mtx` and `cvar`